File: app/lib/data/dto/team_message.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'team_message.freezed.dart';
part 'team_message.g.dart';

@freezed
class TeamMessage with _$TeamMessage {
  factory TeamMessage({
    required String userId,
    required String teamId,
    required String text,
    required DateTime date,
  }) = _TeamMessage;

  factory TeamMessage.fromJson(Map<String, dynamic> json) =>
      _$TeamMessageFromJson(json);
}

File: server/models/teamMessage.js
import mongoose from 'mongoose'

const TeamMessage = new mongoose.Schema({
        userId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'User',
            required: true,
        },
        teamId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Team',
            required: true,
        },
        text: {
            type: String,
            required: true,
        },
        date: {
            type: Date,
            required: true,
        }
    },
    {
        versionKey: false,
    }
);

export default mongoose.model('TeamMessage', TeamMessage);
